lib/cfg.c

```diff
@@ -368,7 +368,9 @@ cfg_init(GlobalConfig *cfg)
   if (!cfg_tree_start(&cfg->tree))
     return FALSE;
 
-  g_assert(cfg_tree_on_inited(&cfg->tree));
+  if (!cfg_tree_on_inited(&cfg->tree))
+    return FALSE;
```

Comment:
Please keep the assertion here. It is intentional, see https://github.com/syslog-ng/syslog-ng/pull/3176#issuecomment-638849597.

TLDR: A half-initialized pipeline turned out to be really hard to deinitialize correctly when dedicated source/destination threads are spawned (because we would have to wait for workers to stop and guarantee some internal task/timer/fdwatch ordering in ivykis during this action).

The best would be to extract these 2 lines or just add a comment.

Reply:
Thanks, I add your comment, and a link to the original comment, those should be a nice point to unroll the shenanigans of this assert.